src/components/SearchBar/SearchBar.tsx

```diff
@@ -37,19 +33,31 @@ const SearchBar: React.ForwardRefRenderFunction<
     disabled = false,
     focused = false,
     label,
-    onClick,
+    error,
+    helperText,
+    onSearch,
     sx,
   } = props;
 
+  const handleChange: ChangeEventHandler<HTMLInputElement> = (event) => {
+    if (onChange != undefined) onChange(event.target.value);
+  };
+
+  const handleClick: MouseEventHandler<HTMLButtonElement> = () => {
+    if (onSearch != undefined) onSearch();
+  };
+
```

Comment:
This "translates" HTML events to simpler function calls for the parent component.